### PR TITLE
Reddit publishers now appear correctly in publisher panel when account is set to hide from robots

### PR DESCRIPTION
**Pull request by jasonrsadler**
Fixes brave/brave-browser#4754

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests && npm run test-security`) on
  - [ ] Windows
  - [ ] macOS
  - [ ] Linux
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:

1. Open Brave with Rewards enabled.
2. Visit a Reddit user page that has 'hide from robots' turned on.
(use reddit.com/u/jsadler-brave as an example of search indexing turned off)
3. Ensure publisher info displays in panel

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: vendor/bat-native-ledger/src/bat/ledger/internal/media/reddit.cc [hunk 1]
```diff
@@ -194,9 +194,10 @@ std::string MediaReddit::GetUserId(const std::string& response) {
   if (response.empty()) {
     return std::string();
   }
-
+  std::string pattern(ExtractData(response, "hideFromRobots\":",
```
Comment (NejcZdovc) [inline vendor/bat-native-ledger/src/bat/ledger/internal/media/reddit.cc]: could you please use `std::string pattern =` and const

File: vendor/bat-native-ledger/src/bat/ledger/internal/media/reddit.cc [hunk 2]
```diff
@@ -194,9 +194,10 @@ std::string MediaReddit::GetUserId(const std::string& response) {
...
+      "\"isEmployee\""));
   std::string id(braveledger_media::ExtractData(
```
Comment (NejcZdovc) [inline vendor/bat-native-ledger/src/bat/ledger/internal/media/reddit.cc]: could you please use `std::string id =`

File: vendor/bat-native-ledger/src/bat/ledger/internal/media/reddit.cc (hunk 1 above)
Comment (jasonrsadler) [reply vendor/bat-native-ledger/src/bat/ledger/internal/media/reddit.cc]: Done

File: vendor/bat-native-ledger/src/bat/ledger/internal/media/reddit.cc (hunk 2 above)
Comment (jasonrsadler) [reply vendor/bat-native-ledger/src/bat/ledger/internal/media/reddit.cc]: Done

File: vendor/bat-native-ledger/src/bat/ledger/internal/media/reddit.cc [hunk 3]
```diff
@@ -194,9 +194,10 @@ std::string MediaReddit::GetUserId(const std::string& response) {
   if (response.empty()) {
     return std::string();
   }
-
-  std::string id(braveledger_media::ExtractData(
-      response, "hideFromRobots\":false,\"id\":\"t2_", "\""));
+  const std::string pattern = ExtractData(response, "hideFromRobots\":",
```
Comment (NejcZdovc) [inline vendor/bat-native-ledger/src/bat/ledger/internal/media/reddit.cc]: nit: `we always use braveledger_media:: ExtractData`

Comment (jasonrsadler) [reply vendor/bat-native-ledger/src/bat/ledger/internal/media/reddit.cc]: Fixed

**Pull request merged**